src/allmydata/scripts/tahoe_add_alias.py

```diff
@@ -111,34 +151,42 @@ def _get_alias_details(nodedir):
     return data
 
 
-def list_aliases(options):
-    nodedir = options['node-directory']
-    stdout = options.stdout
-    stderr = options.stderr
+def _escape_format(t):
+    """
+    _escape_format(t).format() == t
 
-    data = _get_alias_details(nodedir)
+    :param unicode t: The text to escape.
+    """
+    return t.replace("{", "{{").replace("}", "}}")
 
-    max_width = max([len(quote_output(name)) for name in data.keys()] + [0])
-    fmt = "%" + str(max_width) + "s: %s"
-    rc = 0
+
+def list_aliases(options):
+    """
+    Show aliases that exist.
+    """
+    data = _get_alias_details(options['node-directory'])
 
     if options['json']:
-        try:
-            # XXX why are we presuming utf-8 output?
-            print(json.dumps(data, indent=4).decode('utf-8'), file=stdout)
-        except (UnicodeEncodeError, UnicodeDecodeError):
-            print(json.dumps(data, indent=4), file=stderr)
-            rc = 1
+        output = _escape_format(json.dumps(data, indent=4).decode("utf-8"))
     else:
-        for name, details in data.items():
-            dircap = details['readonly'] if options['readonly-uri'] else details['readwrite']
-            try:
-                print(fmt % (unicode_to_output(name), unicode_to_output(dircap.decode('utf-8'))), file=stdout)
-            except (UnicodeEncodeError, UnicodeDecodeError):
-                print(fmt % (quote_output(name), quote_output(dircap)), file=stderr)
-                rc = 1
-
-    if rc == 1:
-        print("\nThis listing included aliases or caps that could not be converted to the terminal" \
-                        "\noutput encoding. These are shown using backslash escapes and in quotes.", file=stderr)
-    return rc
+        def dircap(details):
+            return (
+                details['readonly']
+                if options['readonly-uri']
+                else details['readwrite']
+            ).decode("utf-8")
+
+        max_width = max([len(quote_output(name)) for name in data.keys()] + [0])
+        fmt = "%" + str(max_width) + "s: %s"
+        output = "\n".join(list(
+            fmt % (name, dircap(details))
```

Comment:
This is ok... but maybe move the formatting inside `dircap` to make the code easier to read for tired people :)